src/frontend/src/views/Data/components/SamplesView/components/SamplesTable/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { CellBasic, CellHeader, Table, TableHeader, TableRow } from "czifui";
+import {
+  createColumnHelper,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { IdMap } from "src/common/utils/dataTransforms";
+import { map } from "lodash";
+import { useEffect, useState } from "react";
+
+// TODO-TR (mlila): types
+interface Props {
+  data: IdMap<Sample> | undefined;
+  isLoading: boolean;
+}
+
+const columnHelper = createColumnHelper<Sample>();
```

Comment:
From what I'm reading in the docs for react-table, they say `createColumnHelper` is intended to be used with the row shape. But looking at the docs closely, I'm not really sure that matters, is it just a type convenience sorta thing? Is it an easy way to have type safety where you're guaranteeing that the row/data object being discussed per-column has all the attributes given? Basically, is the provided type more about the data structure or the row structure? If it's the case that it's about data more, then I think `createColumnHelper<Sample>` is totally fine and reasonable. But if it actually ties in more deeply to the row shape than the data shape, it feels like we need another `SampleRow` type or something.

Reply:
https://app.shortcut.com/genepi/story/227054